src/history.rs

```diff
@@ -1,72 +1,72 @@
-        )
-    }
-}
 
-/// Do a logical clock comparison, ignoring the source probe_id
-impl PartialEq for CausalSnapshot {
-    fn eq(&self, other: &Self) -> bool {
-        self.clock == other.clock
-    }
-}
+use fixed_slice_vec::{
+    single::{EmbedValueError, SplitUninitError},
+    FixedSliceVec, TryPushError,
+};
+use race_buffer::writer::OverwrittenEntry;
+use static_assertions::{assert_eq_align, assert_eq_size, const_assert, const_assert_eq};
 
-impl PartialOrd for CausalSnapshot {
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        self.clock.partial_cmp(&other.clock)
-    }
-}
+use crate::{
+    log::{LogEntry, RaceLog},
+    payload::{write_report_payload, FrontierClocksFSV, PayloadOutputByteSlice},
+    wire::{report::WireReport, WireCausalSnapshot},
+    CausalSnapshot, EventId, LogicalClock, MergeError, ModalityProbeInstant, OrdClock, ProbeEpoch,
+    ProbeId, ProbeTicks, ProduceError, ReportError, StorageSetupError,
+};
 
 pub const MIN_CLOCKS_LEN: usize = 2;
 pub const MIN_LOG_LEN: usize = MIN_CLOCKS_LEN * 16;
 pub const MIN_HISTORY_SIZE_BYTES: usize = size_of::<DynamicHistory>()
     + 3 * size_of::<u32>()
     + MIN_CLOCKS_LEN * size_of::<LogicalClock>()
-    + MIN_LOG_LEN * size_of::<CompactLogItem>();
-
-const EPOCH_WRAPAROUND_THRESHOLD_TOP: u16 = ProbeEpoch::MAX - 3;
-const EPOCH_WRAPAROUND_THRESHOLD_BOTTOM: u16 = 3;
+    + MIN_LOG_LEN * size_of::<LogEntry>();
 
+#[cfg(target_pointer_width = "32")]
+const_assert_eq!(align_of::<u64>(), align_of::<DynamicHistory>());
+#[cfg(not(target_pointer_width = "32"))]
 const_assert_eq!(align_of::<usize>(), align_of::<DynamicHistory>());
+
 const_assert_eq!(4, align_of::<LogicalClock>());
-const_assert_eq!(4, align_of::<CompactLogItem>());
+const_assert_eq!(4, align_of::<LogEntry>());
 
 assert_eq_size!(u64, LogicalClock);
 assert_eq_align!(u32, LogicalClock);
 
-assert_eq_size!(u32, CompactLogItem);
-assert_eq_align!(u32, CompactLogItem);
+assert_eq_size!(u32, LogEntry);
+assert_eq_align!(u32, LogEntry);
 
 const_assert_eq!(12, size_of::<CausalSnapshot>());
 const_assert_eq!(4, align_of::<CausalSnapshot>());
 
 const_assert_eq!(12, size_of::<ModalityProbeInstant>());
 const_assert_eq!(4, align_of::<ModalityProbeInstant>());
 
+#[cfg(target_pointer_width = "32")]
 const_assert_eq!(
-    10 + size_of::<FixedSliceVec<'_, LogicalClock>>()
-        + size_of::<CompactLogVec<'_>>()
-        + size_of::<ChunkedReportState>(),
+    size_of::<ProbeId>()
+        + size_of::<u32>()
+        + size_of::<FixedSliceVec<'_, LogicalClock>>()
+        + size_of::<LogicalClock>()
+        + size_of::<RaceLog<'_>>()
+        + size_of::<usize>()
+        + size_of::<u64>(),
+    size_of::<DynamicHistory>()
+);
+
+#[cfg(not(target_pointer_width = "32"))]
+const_assert_eq!(
+    4 + size_of::<ProbeId>()
+        + size_of::<u32>()
+        + size_of::<FixedSliceVec<'_, LogicalClock>>()
+        + size_of::<LogicalClock>()
+        + size_of::<RaceLog<'_>>()
+        + size_of::<u64>()
```

Comment:
It looks like the difference between these two is a pointer width kind of difference (usize when target_pointer_width = 32, u64 otherwise [and otherwise probably means 64-bit]), so I'm curious why using size_of::<usize> for this part of the equation doesn't work for both scenarios.